extensions/spring-data-jpa/deployment/src/main/java/io/quarkus/spring/data/deployment/generate/CustomQueryMethodsAdder.java

```diff
@@ -0,0 +1,271 @@
+
+                        if (DotNames.VOID.equals(methodReturnTypeDotName)) {
+                            methodCreator.returnValue(null);
+                        }
+                        handleLongReturnValue(methodCreator, deleteCount, methodReturnTypeDotName);
+
+                    } else if (queryString.startsWith("update") || queryString.startsWith("UPDATE")) {
+                        if (!DotNames.PRIMITIVE_INTEGER.equals(methodReturnTypeDotName)
+                                && !DotNames.INTEGER.equals(methodReturnTypeDotName)
+                                && !DotNames.VOID.equals(methodReturnTypeDotName)) {
+                            throw new IllegalArgumentException(
+                                    method.name() + " of Repository " + repositoryClassInfo
+                                            + " is meant to be an update query and can therefore only have a void or integer return type");
+                        }
+
+                        ResultHandle updateCount;
+                        if (useNamedParams) {
+                            ResultHandle parameters = generateParametersObject(namedParameterToIndex, methodCreator);
+                            ResultHandle parametersMap = methodCreator.invokeVirtualMethod(
+                                    MethodDescriptor.ofMethod(Parameters.class, "map", Map.class),
+                                    parameters);
+
+                            // call JpaOperations.executeUpdate
+                            updateCount = methodCreator.invokeStaticMethod(
+                                    MethodDescriptor.ofMethod(JpaOperations.class, "executeUpdate", int.class,
+                                            String.class, Map.class),
+                                    methodCreator.load(queryString), parametersMap);
+                        } else {
+                            ResultHandle paramsArray = generateParamsArray(queryParameterIndexes, methodCreator);
+
+                            // call JpaOperations.executeUpdate
+                            updateCount = methodCreator.invokeStaticMethod(
+                                    MethodDescriptor.ofMethod(JpaOperations.class, "executeUpdate", int.class,
+                                            String.class, Object[].class),
+                                    methodCreator.load(queryString), paramsArray);
+                        }
+
+                        if (DotNames.VOID.equals(methodReturnTypeDotName)) {
+                            methodCreator.returnValue(null);
+                        }
+                        handleIntegerReturnValue(methodCreator, updateCount, methodReturnTypeDotName);
+
+                    } else {
+                        throw new IllegalArgumentException(
+                                method.name() + " of Repository " + repositoryClassInfo
+                                        + " has been annotated with @Modifying but the in @Query appears to be " +
```

Comment:
```suggestion
                                        + " has been annotated with @Modifying but the @Query does not appear to be " +
```
Looked like the exact opposite of what you wanted to say or am I missing something?

Reply:
Great catch!